sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -560,73 +559,65 @@ public static void removeAllControlBlocksAndDatasetsAndExtRefSrcBindings(final S
         SubstationAdapter substationAdapter = sclRootAdapter.getSubstationAdapter();
         final List<Pair<String, String>> iedNameLdInstList = substationAdapter.getIedAndLDeviceNamesForLN0FromLNode();
-        List<SclReportItem> errors = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .flatMap(IEDAdapter::streamLDeviceAdapters)
                 .map(LDeviceAdapter::getLN0Adapter)
                 .map(ln0Adapter -> ln0Adapter.updateLDeviceStatus(iedNameLdInstList))
                 .flatMap(Optional::stream)
-                .toList();
-        SclReport sclReport = new SclReport();
-        sclReport.getSclReportItems().addAll(errors);
-        sclReport.setSclRootAdapter(sclRootAdapter);
-        return sclReport;
+                .collect(Collectors.toList());
     }
 
     /**
      * Checks Control Blocks, DataSets and FCDA number limitation into Access Points
      *
      * @param scd SCL file for which LDevice should be activated or deactivated
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport analyzeDataGroups(SCL scd) {
+    public static List<SclReportItem> analyzeDataGroups(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        List<SclReportItem> sclReportItems = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .map(iedAdapter -> {
                     List<SclReportItem> list = new ArrayList<>();
                     list.addAll(iedAdapter.checkDataGroupCoherence());
                     list.addAll(iedAdapter.checkBindingDataGroupCoherence());
                     return list;
-                }).flatMap(Collection::stream).toList();
-
-        return new SclReport(sclRootAdapter, sclReportItems);
+                }).flatMap(Collection::stream).collect(Collectors.toList());
     }
 
     /**
      * Update DAIs of DO InRef in all LN0 of the SCD using matching ExtRef information.
      *
      * @param scd SCL file for which DOs InRef should be updated with matching ExtRef information
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport updateDoInRef(SCL scd) {
+    public static List<SclReportItem> updateDoInRef(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        List<SclReportItem> sclReportItems = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .flatMap(IEDAdapter::streamLDeviceAdapters)
                 .map(LDeviceAdapter::getLN0Adapter)
                 .map(LN0Adapter::updateDoInRef)
                 .flatMap(List::stream)
-                .toList();
-        return new SclReport(sclRootAdapter, sclReportItems);
+                .collect(Collectors.toList());
```

Comment:
Use `.toList()`. Caller should not assume that the returned List is modifiable.

Reply:
done